integration-tests/src/tests/runtime/state_viewer.rs

```diff
@@ -12,11 +16,89 @@ use near_primitives::{
     version::PROTOCOL_VERSION,
 };
 use near_primitives_core::serialize::from_base64;
-use near_store::set_account;
+use near_store::{set_account, NibbleSlice, RawTrieNode, RawTrieNodeWithSize};
 use node_runtime::state_viewer::errors;
 use node_runtime::state_viewer::*;
 use testlib::runtime_utils::{alice_account, encode_int};
 
+struct ProofVerifier {
+    nodes: HashMap<CryptoHash, RawTrieNodeWithSize>,
+}
+
+impl ProofVerifier {
+    fn new(proof: Vec<Arc<[u8]>>) -> Self {
+        let nodes = proof
+            .into_iter()
+            .map(|bytes| {
+                let hash = CryptoHash::hash_bytes(&bytes);
+                let node = RawTrieNodeWithSize::decode(&bytes).unwrap();
```

Comment:
We probably want to return the error here.  If the proof contains invalid nodes than that’s invalid proof rather than a crash.